File: rekkon.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import sys
import os
from subprocess import call

class Color:
	PURPLE = '\033[95m'
	RED = '\033[91m'
	BRIGHTGREEN = '\033[40m'
	BOLD = '\033[1m'
	END = '\033[0m'

def submenu():
	selection = raw_input(Color.PURPLE + "Go Back to Main Menu [y/N]:" + Color.END)
	if selection == "y":
		main_menu()
	else:
		print "Exiting..."
		sys.exit(1)

def main():
	if len(sys.argv) < 2:
		print "Usage: michellerecon.py localhostip remotehostip"
		print "Exiting..."
		sys.exit(1)
	else:
		main_menu()

def main_menu_design():
	call('clear')	
	print "-------------------------------"
	print "|            Rekkon           |" 
	print "-------------------------------"
	print "Mishell Security @mishellsec"
	print "       .---."
	print "      /     \\"
	print "      \.@-@./"
	print "      /`\_/`\\"
	print "     //  _  \\"
	print "    | \\     )|_"
	print "   /`\\_`>  <_/ \\"
	print "   \\__/'---'\\__/"
	print "\n-------------------------------"
	print "Local Host IP: " + sys.argv[1]	
	print "Remote Host IP: " + sys.argv[2]
	print "-------------------------------"
	

def main_menu():
   	try:
		main_menu_design()
		localhostip = sys.argv[1]
		remotehostip = sys.argv[2]
		print " 1 - Nmap " 
		print " 2 - Nikto " 
		print " 3 - Gobuster " 
		print " 4 - Handler linux/x86 "
		print " 5 - Handler windows/reverse_tcp"
		print " 6 - Searchsploit"
		print " 7 - Searchsploit -p"
		print " 8 - enum4linux"
		print " 9 - wireshark"
		print "10 - davtest " + remotehostip
		print "11 - Nmap [port]"
		print "12 - Open in Firefox " + remotehostip
		print "13 - Open in Firefox [port]"
		print "14 - ifconfig"	
		print "15 - cadaver"		
		print "16 - Zombie Scan"
		print "17 - Netcat Listener"	
		print "18 - Stop Apache"
		print "19 - Start Apache"	
		print "20 - SMB Enum Shares"
		print "--------------------------------"
		print "99 - More Options"
		print " 0 - Exit\n"	
		selection = int(raw_input(Color.PURPLE + "Enter your selection: " + Color.END))

		if selection == 1:
			call(["nmap", "-Pn", "-sC", "-sV", remotehostip])						
		elif selection == 2: 
			call(["nikto", "-h", remotehostip])						
		elif selection == 3: 
			call(["gobuster", "-w", "/usr/share/wordlists/dirbuster/directory-list-2.3-medium.txt", "-t", "20", "-u", "http://" + remotehostip])			
		elif selection == 4:
			lport = raw_input("Enter Local Port to use: ")
			os.system("msfconsole -qx \"use exploit/multi/handler;set payload linux/x86/shell/reverse_tcp;set LPORT " + lport + ";set LHOST " + localhostip +";exploit\"")			
		elif selection == 5:
			lport = raw_input("Enter Local Port to use: ")
			os.system("msfconsole -qx \"use exploit/multi/handler;set payload windows/meterpreter/reverse_tcp;set LPORT " + lport + ";set LHOST " + localhostip +";exploit\"")			
		elif selection == 6: 
			searchterm = raw_input("Enter search term: ")
			call(["searchsploit", searchterm])			
		elif selection == 7: 
			searchterm = raw_input("Enter exploit id to copy: ")
			call(["searchsploit", "-p", searchterm])			
		elif selection == 8:
			call(["enum4linux", remotehostip])			 
		elif selection == 9:
			call(["sudo", "wireshark"])			 
		elif selection == 10:		
			call(["davtest", "-url", "http://" + remotehostip])			
		elif selection == 11:
			port = raw_input("Enter port to scan: ")
			call(["nmap", "-Pn", "-sC", "-sV", "-p", port, remotehostip])
		elif selection == 12:
			call(["firefox", remotehostip])	
		elif selection == 13:
			port = raw_input("Enter port for ip: ")
			call(["firefox", remotehostip + ":" + port])
		elif selection == 14:
			call(["ifconfig"])		
		elif selection == 15:
			call(["cadaver", remotehostip])
		elif selection == 16:
			port = raw_input("Enter port to use for scan: ")
			call(["hping3", "-s", "-r", "-p", port, remotehostip])
		elif selection == 17:
			port = raw_input("Enter port to listen on: ")
			call(["nc", "-l", "-v", "-p", port])
		elif selection == 18:
			call(["service", "apache", "stop"])
		elif selection == 19:
			call(["service", "apache", "start"])
		elif selection == 20:
			os.system("msfconsole -qx \"use auxiliary/scanner/smb/smb_enumshares;set RHOSTS " + remotehostip + ";run;exit\"")
		elif selection == 99:
			menu20()			
		else:
			sys.exit(1)

		submenu()

     	except:
	  	sys.exit(1) 

def menu20():
	try:
		main_menu_design()
		localhostip = sys.argv[1]
		remotehostip = sys.argv[2]
		print "21 - Test " 
		
		print "--------------------------------"
		print "98 - Main Menu"
		#print "99 - More Options"
		print " 0 - Exit\n"	
		selection = int(raw_input(Color.PURPLE + "Enter your selection: " + Color.END))
		if selection == 21:
			print "Test 21"			
		elif selection == 98:
			main_menu()			
		else:
			sys.exit(1)
		submenu()
	except:
		sys.exit(1) 

main()

File: rekkon.rb
#!/usr/bin/ruby
$lhost = "127.0.0.1"
$rhost = "192.168.1.1"
$lport = "4444"
$rport = "80"

class String
  def red; "\e[31m#{self}\e[0m" end
  def green; "\e[32m#{self}\e[0m" end
  def yellow; "\e[33m#{self}\e[0m" end
  def blue; "\e[34m#{self}\e[0m" end
  def magenta; "\e[35m#{self}\e[0m" end
  def cyan; "\e[36m#{self}\e[0m" end
end

def submenu
	print "Go Back to Main Menu [y/N]:".red
  selection = gets.strip
  case selection
    when "y"
      main_menu
    when "Y"
      main_menu
    when "yes"
      main_menu
    else
      print "Exiting..."
      exit(0)
    end
end

def main
  print "Enter the local host IP: ".red
  $lhost = gets
  print "Enter the remote host IP: ".red
  $rhost = gets 
  main_menu
end

def main_menu_design
  localhost = "\nLocal Host IP: " << $lhost
  remotehost = "\nRemote Host IP: " << $rhost
	system "clear"
	print "\n-------------------------------"
	print "\n|            Rekkon           |" 
	print "\n-------------------------------"
	print "\nMishell Security @michellesec".magenta
	print "\n       .---.".yellow
	print "\n      /     \\".yellow
	print "\n      \\.@-@./".yellow
	print "\n      /`\\_/`\\\\".yellow
	print "\n     //  _  \\\\".yellow
	print "\n    | \\\\     )|_".yellow
	print "\n   /`\\\\_`>  <_/ \\\\".yellow
	print "\n   \\\\__/'---'\\\\__/".yellow
	print "\n-------------------------------"
	print localhost.red
  print remotehost.red
	print "-------------------------------\n"
end	

def main_menu
    main_menu_design
    print " 1 - Nmap\n" 
  	print " 2 - Nikto\n" 
  	print " 3 - Dirbuster\n" 
  	print " 4 - Metasploit"
  	print " 5 - Ftp"
  	print " 6 - Searchsploit\n"
  	print " 7 - Searchsploit -p\n"
  	print " 8 - enum4linux\n"
  	print " 9 - wireshark\n"
  	print "10 - davtest " + $rhost
  	print "11 - Nmap [port]\n"
  	print "12 - Open in Firefox " + $rhost
  	print "13 - Open in Firefox [port]\n"
  	print "14 - ifconfig\n"	
  	print "15 - cadaver\n"		
  	print "16 - Zombie Scan\n"
  	print "17 - Netcat Listener\n"	
  	print "18 - Stop Apache\n"
  	print "19 - Start Apache\n"	
  	print "20 - nmblookup\n" 	
  	print " 0 - Exit\n"	
  	print "Enter your selection: ".red
    selection = gets.strip
    case selection
      when "1"
        system "nmap -Pn -sC -sV " + $rhost
      when "2"
        system "nikto -h " + $rhost
      when "3"
        system "dirb " + $rhost
      when "4"
        system "msfconsole"
      when "5"
        system "ftp " + $rhost
      when "6"
        print "Enter Search term: "
        searchterm = gets
        system "searchsploit " + searchterm
      when "7"
        print "Enter exploit id to copy: "
        searchterm = gets
        system "searchsploit -p " + searchterm
      when "8"
        system "enum4linux " + $rhost
      when "9"
        system "sudo wireshark"
      when "10"
        system "davtest " + $rhost
      when "11"
        print "Enter port to scan: "
        port2 = gets.strip
        system "nmap -Pn -sC -sV -p" + port2 + " " + $rhost 
      when "12"
        system "firefox " + $rhost
      when "13"
        print "Enter port to scan: "
        port2 = gets.strip
        system "firefox " + $rhost + ":" + port2
      when "14"
        system "ifconfig"
      when "15"
        system "cadaver " + $rhost
      when "16"
        print "Enter port to use for scan: "
        port2 = gets.strip
        system "hping3 -s -r -p " + port2 + " " + $rhost
      when "17"
        print "Enter port to listen on: "
        port2 = gets.strip
        system "nc -l -v -p " + port2
      when "18"
        system "service apache stop"
      when "19"
        system "service apache start"
      when "20"
        system "nmblookup -A " + $rhost
      else
        exit(0)
    end
    
    submenu
	
  
  
  
end

main